src/renderer/vt/Xterm256Engine.cpp

```diff
@@ -128,6 +131,33 @@ Xterm256Engine::Xterm256Engine(_In_ wil::unique_hfile hPipe,
     return S_OK;
 }
 
+// Routine Description:
+// - Write a VT sequence to start/stop a hyperlink
+// Arguments:
+// - textAttributes - Text attributes to use for the hyperlink ID
+// - pData - The interface to console data structures required for rendering
+// Return Value:
+// - S_OK if we succeeded, else an appropriate HRESULT for failing to allocate or write.
+HRESULT Microsoft::Console::Render::Xterm256Engine::_UpdateHyperlinkAttr(const TextAttribute& textAttributes,
+                                                                         const gsl::not_null<IRenderData*> pData) noexcept
+{
+    if (textAttributes.IsHyperlink() != _lastTextAttributes.IsHyperlink())
+    {
+        if (textAttributes.IsHyperlink())
+        {
+            USHORT id = textAttributes.GetHyperlinkId();
+            RETURN_IF_FAILED(_SetHyperlink(pData->GetHyperlinkUri(id), id));
+        }
+        else
+        {
+            RETURN_IF_FAILED(_SetHyperlink(L"", 0));

```

Comment:
This is where I'm thinking that `EndHyperlink` might be cleaner - no need to remember that these special params mean "we're ending the hyperlink"

Reply:
Again, resolving here and leaving the comment thread above for where we discuss this